### ISSUE TITLE
Callisto TipBot twitter integration

### ISSUE THREAD

**Issue by Exzender** [EthereumCommonwealth/Proposals#89]
Callisto TipBot twitter integration.

Task: get bot to monitor retweet activity for selected post in Twitter - for further AirDrops (Rains) in Telegram group only to those Users who really retweeted the post. It should increase users twitter activity.
 
Problem: Twitter API has limitation: only 100 latest retweets can be returned (so Bot can not check ALL retweets just before the Rain). There is a walkaround - bot needs to monitor tweet and collect retweets as they appear. And this can be implemented in different ways. 

What way to choose?
And should any chat/group user be able to use such tweet promotion or should it be limited to group admins only?

Version 1st (simple one) (can be implemented faster)
1. Team or user provides tweet link to the bot 
smthng like: /tweet http://link
2. Bot collects info
3. When Team decides that there are enough retweets - they call rain providing link to the tweet
/rain 10000 all http://link
4. Bot now checks not only users "registered" in it's DB, but finds them in collected retweets
5. If step 1 was not perfomed - bot still can check latest 100 retweets - but in this case some users who retweeted may be missed

Version 2nd (full automatization) (more time for implementation, bot has to much control...)
1. Team provides tweet link to the bot and retweets target and value to rain
smthng like: /tweet http://link 150 10000
1.1. Bot transfers 10000 clo to special wallet
2. Bot collects retweets info
3. From time-to-time bot sends message to chat with twitter link and current number of retweets
4. When bot finds that there are enough retweets - it stops counting and automatically fires the rain from special wallet
5. Step similar to step 4 in Version 1st
6. Also there can be some deadline (if we did not reach target retweets - collection stopped and funds returned to source wallet)
7. Or can be added some kind of command - to stop tweet monitoring

Version 3rd (semi-automatization)
Some mix of 1st and 2nd versions

**Comment by shivneja** [comment]
Great idea 👍.

This is something we should definitely launch. AFAIK, Twitter is actively moving towards crypto integration so if we have a functional bot it would be awesome.

Few points:
1. Anyway we can monitor all tweets which have $CLO and/or #callistonetwork in the text? We can tip (say) XXX for every 10 Tweets (this is tweets and not RT or Likes)
2. Instead of the link you have mentioned can we use $CLO or #callistnetwork in the RTs as a counter?
3. For your question above I would say choose option 1 and get it started and then keep working on refining it and integrating into Option 2.

**Comment by ghost** [comment]
I would just like to agree with this proposal as the activity of the promotions would increase a lot now when users have to engage to be able to win! 

Question for @Exzender :  You say it is only 100 latest retweets, what about comments? 
personally i feel commenting should be the way for the user to join the pool of people who take part of the rain

We can still have the rule:  Rain will be activated once we reach 100 RT ( for example)
But they have to comment to win

**Comment by shivneja** [comment]
@Spigfish great suggestion. 

Maybe better to implement in stages? First stage just make a good workable product calculating RTs only, then next stage we make sure there are comments and only the RTs with comments are counted for rain or in second stage if workable, give a higher proportion of the rain?? So we reward RTs with value X and we reward RTs with comments with value 6X (say).

**Comment by ghost** [comment]
> @Spigfish great suggestion.
> 
> Maybe better to implement in stages? First stage just make a good workable product calculating RTs only, then next stage we make sure there are comments and only the RTs with comments are counted for rain or in second stage if workable, give a higher proportion of the rain?? So we reward RTs with value X and we reward RTs with comments with value 6X (say).

True, If it is possible , this proposal would be nice to get active asap and then work on improvements! :)

**Comment by shivneja** [comment]
@Spigfish @Exzender 

I have seen some of the other channels and they are using non-native coins for tipping. So for example, a token on ETH is using TRX for tipping. 

Based on the suite of products that you have developed I.e. tip bot + giveaway + rain + lottery + cold staking + future developments linked to #90 and #85 and #86 there might be potential to sell this to other channels which are struggling with keeping interest going. CLO as a tip coin is very attractive because 100 CLO sounds better as a tip than 0.000494 ETH or 0.00001327 BTC. 

CLO is a very strong community (and crypto universe recognises that) and maybe you would like to leverage that by selling some of the tools as a package of services.

**Comment by Exzender** [comment]
Great idea!  @shivneja  can You suggest channels ? Can You DM me (@kutuk) in Telegram ?

**Comment by shivneja** [comment]
@Exzender since you work at Lightning speed can you take a look at this 
https://breakermag.com/what-is-tippn-me-and-how-to-make-money-off-it-a-journey/

In essence, tippin.me is currently the twitter tipping favourite and uses Lightning to hand out Bitcoin tips. What’s another way that we can get tips distributed on twitter? How can you integrate your tipping operation into Twitter?

**Comment by ghost** [comment]
> @Exzender since you work at Lightning speed can you take a look at this
> https://breakermag.com/what-is-tippn-me-and-how-to-make-money-off-it-a-journey/
> 
> In essence, tippin.me is currently the twitter tipping favourite and uses Lightning to hand out Bitcoin tips. What’s another way that we can get tips distributed on twitter? How can you integrate your tipping operation into Twitter?

https://twitter.com/cctip_io
is a nice one also

**Comment by Exzender** [comment]
There is a whole Coin develepod only for Tipping in twitter: https://twit.rocks/  :)
There are many tipping bots already...
As for our bot - I like the way it's released in tipping.me - through the extension. But as always it as PROs and CONs. I'm in the process of studying.
